src/components/pill-selector-dropdown/PillSelectorDropdown.js

```diff
@@ -177,21 +176,20 @@ class PillSelectorDropdown extends React.Component<Props, State> {
     };
 
     handleEnter = (event: SyntheticEvent<>) => {
-        const { isInCompositionMode } = this.state;
+        const { isInCompositionMode, inputValue } = this.state;
         if (!isInCompositionMode) {
             event.preventDefault();
-            this.addPillsFromInput();
+            this.addPillsFromInput(inputValue);
         }
     };
 
-    handlePaste = () => {
-        /**
-         * NOTE (ishay): setTimeout is necessary because
-         * otherwise addPillsFromInput gets triggered as soon
-         * as the user "paste's", but before the inputValue
-         * is actually updated.
-         */
-        setTimeout(this.addPillsFromInput, 0);
+    handlePaste = (event: SyntheticClipboardEvent<HTMLInputElement>) => {
+        event.nativeEvent.preventDefault();
+
+        const inputValue: string = event.clipboardData.getData('Text');
+        this.setState({ inputValue });
+        this.props.onInput(inputValue, event);
```

Comment:
Is this going to cause issues with consuming expecting a `SyntheticInputEvent` rather than a `SyntheticClipboardEvent`? The flow type currently doesn't specify the parameters.

Reply:
The types around this are unfortunate :( BUT, the type for `handleInput` specifies an event or some object with `target`, which `SyntheticClipboarEvent` – and by extension its native cousin – has. So the minimum is checked here.